### PR TITLE
#25: Fail build on warning

### LINKED ISSUE #25: Fail build on warning
## Background
Improve project settings.

## Feature
Fail build on warning to ensure no deprecated cod or suspicious construction is used.

### PR DESCRIPTION
**Pull request by benedeki**
* `build.sbt` changed to fail on compilation warning
* other `build.sbt` improvements
* `build.yml` now includes the `doc` stage for better/safer CICD

Closes #25

### CODE REVIEW

**Comment by miroslavpojer** [comment]
- [ ] test run

tested commands on local, all failing, see build failures on build server

- sbt ++2.11.12 test doc
- sbt ++2.12.12 test doc

**Comment by miroslavpojer** [comment]
- [ ] Test run

**Review by miroslavpojer — APPROVED** [review]
- [x] code review
- [x] pulled
- [x] built

File: build.sbt [hunk 1]
```diff
@@ -14,19 +14,22 @@
  * limitations under the License.
  */
 
-ThisBuild / organization := "za.co.absa.fa-db"
+import Dependencies._
+
+ThisBuild / name := "fa-db"
+ThisBuild / organization := "za.co.absa"
```
Comment (Zejnilovic) [inline build.sbt:20]: Will this not change the path from `za.co.absa.fa-db` to `za.co.absa`? 
So paths will ow be `za.co.absa.core_2.12` and `za.co.absa.slick_2.12`

Comment (benedeki) [reply build.sbt:20]: You are right. 👍  Actually not sure, how the change slipped in. 😊

**Review by lsulak — APPROVED** [review]
LGTM

**Review by miroslavpojer — APPROVED** [review]
- [x] pulled
- [x] built
- [x] test run
- [ ] code review

**Review by miroslavpojer — DISMISSED** [review]

**Review by lsulak — DISMISSED** [review]

**Review by lsulak — APPROVED** [review]

**Comment by benedeki** [comment]
Code coverage increased.

**Pull request merged**